odoo/orm/query.py

```diff
@@ -113,6 +116,16 @@ def add_join(self, kind: str, alias: str | TableSQL, table: str | SQL | None, co
         self._joins[alias] = (sql_kind, table, condition)
         self._ids = self._ids and None
 
+    def add_cte(self, alias: str, cte: SQL, recursive: bool = False, materialized: bool | None = None):
```

Comment:
Flags should always be given by keyword.
```suggestion
    def add_cte(self, alias: str, cte: SQL, *, recursive: bool = False, materialized: bool | None = None):
```